src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -199,15 +199,14 @@
   "Returns a boolean indicating whether or not the given certname exists in the db"
   [certname]
   {:pre [certname]}
-  (sql/with-query-results result-set
-    ["SELECT 1 FROM certnames WHERE certname=? LIMIT 1" certname]
-    (pos? (count result-set))))
+  (not (empty? (jdbc/query ["SELECT 1 FROM certnames WHERE certname=? LIMIT 1"
```

Comment:
@ajroetker I take that back, the default for as-arrays? is false, so it calls doall on the result, not vec.